### PR TITLE
Add Price filter for French

### PR DESCRIPTION
**Pull request by rlustin**
@Signez, could you please review?

### CODE REVIEW

File: src/filters/price_filter.rs [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    fn message(&self) -> &'static str {
+        match &self.locale as &str {
+            "fr" => "The currency sign should be written after the amount and a non-breaking space.",
+            _ => "",
+        }
+    }
+
+    fn regex_pattern(&self) -> String {
+        match &self.locale as &str {
+            "fr" => format!("([\\d]+[^ ][{}]{{1}}|[{}]{{1}}[\\s]?[\\d]+)", self.currencies(), self.currencies()),
```
Comment (Signez) [inline src/filters/price_filter.rs]: For future developers sanity, I think this line would benefit from a full-line comment above it, showing example of what it will produce (even parts of it).

File: src/filters/mod.rs [hunk 2]
```diff
@@ -42,5 +44,5 @@ pub trait LinterFilter {
     }
 
     fn message(&self) -> &'static str;
-    fn regex_pattern(&self) -> &'static str;
+    fn regex_pattern(&self) -> String;
```
Comment (Signez) [inline src/filters/mod.rs:47]: Shouldn't those changes be made in a separate commit?

**Review by Signez — CHANGES_REQUESTED** [review]
A few changes needed, but mainly okay. Good job!

File: src/filters/mod.rs (hunk 2 above)
Comment (rlustin) [reply src/filters/mod.rs:47]: You are right!

File: src/filters/price_filter.rs (hunk 1 above)
Comment (rlustin) [reply src/filters/price_filter.rs]: Isn’t the purpose of tests in the same file?

Comment (Signez) [reply src/filters/price_filter.rs]: Not really; in my opinion, the provided tests help to understand the meaning of the filter, but not the "final" content of this regex. We could create tests for this individual function (that tests a fragment of it results, for exemple?) to help us understand what this regex looks like.

This is mainly the problem of complex regex: they are rarely ligitible. An English-written explanation could also helps a lot to dive into it, like `// Checks for digits followed by non-spaces, then by any of currencies() return value`… would be enough imho.

Comment (rlustin) [reply src/filters/price_filter.rs]: What’s about:
```rust
fn regex_pattern(&self) -> String {
    match &self.locale as &str {
        // Matches one of the following:
        // - digits followed by something other than a non-breaking space followed by any of
        //   currencies() return values (ex: `120€` or `120 $`);
        // - any of currencies() return values followed by any whitespace character (or not)
        //   followed by digits (ex: `€120` or `$ 120`).
        "fr" => format!("([\\d]+[^ ][{}]{{1}}|[{}]{{1}}[\\s]?[\\d]+)", self.currencies(), self.currencies()),
        _ => "".to_string(),
    }
}
```

Comment (Signez) [reply src/filters/price_filter.rs]: That would be really good. 👍

Comment (rlustin) [reply src/filters/price_filter.rs]: Fixed then.

**Review by Signez — APPROVED** [review]
